### PR TITLE
sha1: rework & formatting

### PR DESCRIPTION
**Pull request by runcom**
Signed-off-by: Antonio Murdaca <runcom@linux.com>

### CODE REVIEW

File: sha1/sha1.go [hunk 1]
```diff
@@ -1,61 +1,65 @@
...
+
+	"github.com/shanemhansen/gossl/sslerr"
+)
 
-//various interface values copied to match crypro/sha1 interface.
+// The size of a SHA1 checksum in bytes.
+const Size = C.SHA_DIGEST_LENGTH
 
+// The blocksize of SHA1 in bytes.
 const BlockSize = 64
```
Comment (vbatts) [inline sha1/sha1.go]: may as well do these in a paren
```golang
const (
    //...
)
```

Comment (runcom) [inline sha1/sha1.go]: done, was following what stdlib is doing w/o parens here, dunno why

**Comment by vbatts** [comment]
more panics removed :+1: 

LGTM

File: sha1/sha1.go [hunk 2]
```diff
@@ -1,61 +1,67 @@
...
+type digest struct {
+	ctx C.SHA_CTX
 }
 
 // New returns a new sha1 hash.Hash
 func New() hash.Hash {
-	h := new(SHA1Hash)
-	if C.SHA1_Init(&h.sha) != 1 {
-		panic("problem creating hash")
+	d := new(digest)
```
Comment (shanemhansen) [inline sha1/sha1.go]: Can you document that we return nil if there's an error?

Comment (runcom) [inline sha1/sha1.go]: sure

Comment (runcom) [inline sha1/sha1.go]: updated, PTAL

**Comment by shanemhansen** [comment]
Thanks @runcom

**Pull request merged**